예외 처리를 안했을 때
Scanner inFile = new Scanner(new File("input.txt")); //처음에 오류 뜬다.

이런식으로 하면 File에 오류가 뜨는 것을 볼 수 있다.

이때 try catch를 이용하면 해결할 수 있다.

//code 19 중요함!!

package Section2;

import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) {
		
		String[] name = new String[1000]; //충분히 크게 잡으면 그만이다.
		String[] number = new String[1000];
		int n = 0;
		
		//배열의 길이가 얼마인지 알기 어렵다.
		
		try {
			Scanner inFile =new Scanner(new File("input.txt")); 
			//처음에 오류 뜨면 어떻게 해결할지 모르기 때문에오류 뜬다.
			
			
			while(inFile.hasNext()) { //detect End of File 더 읽을게 있으면 계속 진행
				name[n] = inFile.next();
				number[n] = inFile.next();
				n++;
			}
			inFile.close();
		
		} catch(FileNotFoundException e) {
			System.out.println("No file");
			System.exit(9);
			
		}
		for(int i=0; i<n; i++)
			System.out.println(name[i] + " : "+ number[i]);
	}
}

